core/src/main/scala/pureconfig/PeriodUtils.scala

```diff
@@ -0,0 +1,61 @@
+package pureconfig
+
+import java.time.Period
+import java.time.temporal.ChronoUnit
+import java.time.temporal.ChronoUnit._
+
+import scala.util.Try
+
+import com.typesafe.config.impl.ConfigImplUtil
+import pureconfig.error.{ CannotConvert, FailureReason }
+
+/**
+ * Utility functions for converting a `String` to a `Period`. The parser accepts the HOCON unit syntax.
+ */
+private[pureconfig] object PeriodUtils {
+
+  /**
+   * Convert a string to a Period while trying to maintain compatibility with Typesafe's abbreviations.
+   */
+  val fromString: String => Either[FailureReason, Period] = { str =>
+    Try(Right(Period.parse(str))).getOrElse(typesafeConfigParsePeriod(str))
+  }
+
+  // This is a copy of the period parser in `com.typesafe.config.impl.SimpleConfig` adapted to be safer (dealing with
+  // `Either` values instead of throwing exceptions). Try not to refactor this too much so as to be easy for anyone
+  // to compare this code with the original.
+  def typesafeConfigParsePeriod(input: String): Either[FailureReason, Period] = {
+    val s = ConfigImplUtil.unicodeTrim(input)
+    val originalUnitString = getUnits(s)
+    var unitString = originalUnitString
+    val numberString = ConfigImplUtil.unicodeTrim(s.substring(0, s.length - unitString.length))
+    var units: ChronoUnit = null
```

Comment:
Yikes. I feel bad about allowing `var` and `null` into here just to be like the typesafe code.

Reply:
Yeah... it's originally Java code and I used IntelliJ to automatically convert it, which I think is the reason why the code looks even uglier than it's Java counterpart.

As the comment says, the reason I just copied the code was to keep the code easy to sync in case the Typesafe Config guys change anything in the parser. I think that the most important thing is to accept every format that the underlying lib accepts out-of-the-box because it eases a lot the adoption of Pureconfig by users using vanilla Typesafe Config.

If it was easy to implement, I'd add Scalacheck tests capturing this compatibility and just refactor this code to something nice. However, it's difficult to capture newly supported formats with naive generators... I think I'll just refactor this code anyway since it's not that complex.

Reply:
Refactored in dcdb3ae. I also changed `DurationUtils` to use milliseconds as the default unit for durations, since it's also assumed by Typesafe Config...